src/NLog.Targets.Seq/LogEventInfoFormatter.cs

```diff
@@ -49,58 +51,87 @@ static LogEventInfoFormatter()
             };
         }
 
-        public static void ToCompactJson(IEnumerable<LogEventInfo> events, TextWriter output, IList<SeqPropertyItem> properties)
+        public static void ToCompactJson(IEnumerable<LogEventInfo> logEvents, TextWriter output, IList<SeqPropertyItem> properties)
         {
-            foreach (var loggingEvent in events)
+            foreach (var logEvent in logEvents)
             {
-                ToJson(loggingEvent, output, properties);
+                ToCompactJson(logEvent, output, properties);
                 output.Write(Environment.NewLine);
             }
         }
 
-        static void ToJson(LogEventInfo loggingEvent, TextWriter output, IList<SeqPropertyItem> properties)
+        public static void ToCompactJson(LogEventInfo logEvent, TextWriter output, IList<SeqPropertyItem> properties)
         {
-            if (loggingEvent == null) throw new ArgumentNullException(nameof(loggingEvent));
+            if (logEvent == null) throw new ArgumentNullException(nameof(logEvent));
             if (output == null) throw new ArgumentNullException(nameof(output));
             if (properties == null) throw new ArgumentNullException(nameof(properties));
 
+            // Without this call, logEventInfo.Properties behaves erratically when templates are used; stll trying
+            // to track down the cause (needs some debugging into NLog).
+            logEvent.FormattedMessage?.ToString();
```

Comment:
@304NotModified have I hit a bug here? Without this line (which I assume triggers some internal message template handling), later calls to `Properties.ContainsKey("X")` return false even with `X` in the template.